astropy/config/configuration.py

```diff
@@ -570,6 +574,57 @@ def get_config(packageormod=None, reload=False, rootname=None):
         return cobj
 
 
+def generate_config(filename=None):
+    """Generates a configuration file, from the list of ConfigItem
+    for each subpackage.
+    """
+
+    # First, we need to import all modules containing a ConfigNamespace
+    # subclass. So this list must be kept up-to-date.
+    modules_to_import = [
```

Comment:
could we import everything and check whether there are any `ConfigNamespace` classes used? 

I'm otherwise afraid that this won't be updated when a new usage is introduced

Reply:
There is no standard place to store `conf` in. The only way is either hardcode or scrape the code.

Reply:
I'm with @bsipocz here. Not sure it is that hard: we'd only have to recursively import modules - e.g., https://stackoverflow.com/questions/3365740/how-to-import-all-submodules
Alternatively, if we somehow put this at the end of a full astropy test run, we can be sure it is complete (anything that is not tested should not be included!)

Reply:
I already have the importing everything hack for a test, I suppose that can be easily reused. I'll dig it up in a bit